### ISSUE TITLE
kafka-bare-quickstart fails with Unsatisfied dependency error

### ISSUE THREAD

**Issue by rsvoboda** [quarkusio/quarkus-quickstarts#881]
kafka-bare-quickstart fails with Unsatisfied dependency error with Quarkus main, running just `mvn clean package -DskipTests` in kafka-bare-quickstart module.

`kafka-bare-quickstart fails with Unsatisfied dependency for type java.util.Map<java.lang.String, java.lang.Object> and qualifiers [@Named(value = "default-kafka-broker")]`

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project kafka-bare-quickstart: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: Found 2 deployment problems:
[ERROR] [1] Unsatisfied dependency for type java.util.Map<java.lang.String, java.lang.Object> and qualifiers [@Named(value = "default-kafka-broker")]
[ERROR] 	- java member: org.acme.kafka.KafkaProviders#config
[ERROR] 	- declared on CLASS bean [types=[org.acme.kafka.KafkaProviders, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.kafka.KafkaProviders]
[ERROR] 	The following beans match by type, but none have matching qualifiers:
[ERROR] 		- Bean [class=java.util.Map, qualifiers=[@Identifier(value = "default-kafka-broker"), @Any]]
[ERROR] [2] Unsatisfied dependency for type java.util.Map<java.lang.String, java.lang.Object> and qualifiers [@Named(value = "default-kafka-broker")]
[ERROR] 	- java member: org.acme.kafka.VertxKafkaProviders#config
[ERROR] 	- declared on CLASS bean [types=[org.acme.kafka.VertxKafkaProviders, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.kafka.VertxKafkaProviders]
[ERROR] 	The following beans match by type, but none have matching qualifiers:
[ERROR] 		- Bean [class=java.util.Map, qualifiers=[@Identifier(value = "default-kafka-broker"), @Any]]
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1106)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:264)
[ERROR] 	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:129)
[ERROR] 	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:413)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] 	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type java.util.Map<java.lang.String, java.lang.Object> and qualifiers [@Named(value = "default-kafka-broker")]
[ERROR] 	- java member: org.acme.kafka.KafkaProviders#config
[ERROR] 	- declared on CLASS bean [types=[org.acme.kafka.KafkaProviders, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.kafka.KafkaProviders]
[ERROR] 	The following beans match by type, but none have matching qualifiers:
[ERROR] 		- Bean [class=java.util.Map, qualifiers=[@Identifier(value = "default-kafka-broker"), @Any]]
[ERROR] 		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:492)
[ERROR] 		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:460)
[ERROR] 		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:252)
[ERROR] 		... 13 more
[ERROR] 	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type java.util.Map<java.lang.String, java.lang.Object> and qualifiers [@Named(value = "default-kafka-broker")]
[ERROR] 	- java member: org.acme.kafka.VertxKafkaProviders#config
[ERROR] 	- declared on CLASS bean [types=[org.acme.kafka.VertxKafkaProviders, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.kafka.VertxKafkaProviders]
[ERROR] 	The following beans match by type, but none have matching qualifiers:
[ERROR] 		- Bean [class=java.util.Map, qualifiers=[@Identifier(value = "default-kafka-broker"), @Any]]
[ERROR] 		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:492)
[ERROR] 		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:460)
[ERROR] 		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:252)
[ERROR] 		... 13 more
```

**Comment by rsvoboda** [comment]
@cescoffier can you check this pls?

**Comment by cescoffier** [comment]
It's because of the change from @Named to @Identified.

**Comment by rsvoboda** [comment]
Thanks, you are quick with the PR, I can scratch my wip changes :)

Can you please link the Quarkus PR?

`javax.inject.Named` => `io.smallrye.common.annotation.Identifier` is sort of vendor lock in, do you thing users / customers will be fine with that?

**Comment by cescoffier** [comment]
I will check for the PR.

We decided to switch to @Identifier because @Named was not correct. It
comes with a certain semantic ***@***.*** can give you the details) that was
incorrect in this case.

**Comment by cescoffier** [comment]
Closed by #882
